File: data/multilang_comparison/program_julia/src/func17.jl
include("program_julia_head.jl")
function func17(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array104 = nothing
   if pCounter > 1
       array104 = vars.data[pCounter]
       pCounter -= 1
       array104.refC += 1
   else
       array104 = JArray(zeros(Int, 269), 269, 1, 104)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array104)
   params0.size += 1
   array105 = func24(params0, get_path(), loopsFactor);
   loopLimit31 = (50)/3 + 1
   for i in 1:loopLimit31 
      for i in 0:(array104.size-1) 
         array104.data[i+1] += 1
      end
   end
   for i in 1:array104.size
       if array104.data[i] == 92
               return array104
      end
   end
   array105.refC -= 1
   if array105.refC == 0
       empty!(array105.data)
       array105 = nothing
   end
   return array104;
end

File: data/multilang_comparison/program_cpp_array/src/func17.cpp
#include "program_cpp_array.hpp" 
Array* func17(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array104;
   if (pCounter > 0) {
      array104 = vars->data[--pCounter];
      array104->refC++;
      DEBUG_COPY(array104->id);
   } else {
      array104 = new Array();
      array104->size = 269;
      array104->refC = 1;
      array104->id = 104;
      array104->data = new unsigned int[array104->size]();
      DEBUG_NEW(array104->id);
   }
   Array_param params0;
   params0.size = 1;
   Array* data_params0[1];
   params0.data = data_params0;
   params0.data[0] = array104;
   Array* array105 = func24(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array105->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   unsigned int loop31 = 0;
   unsigned int loopLimit31 = (50)/3 + 1;
   for(; loop31 < loopLimit31; loop31++) {
      for (int i = 0; i < array104->size; i++) {
         array104->data[i]++;
      }
   }
   for (int i = 0; i < array104->size; i++) {
       if (array104->data[i] == 92) {
           return array104;
       }
   }
   array105->refC--;
   if(array105->refC == 0) {
      delete[] array105->data;
      array105->data = nullptr;
      DEBUG_FREE(array105->id);
      delete array105;
   }
   return array104;
}

